pkg/controller/deployment/deployment_controller.go

```diff
@@ -170,6 +178,23 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileDeployment) removeSidecar(ctx context.Context, dep *appsv1.Deployment) {
```

Comment:
Is this the same logic that we used for the namespace case? 

Is there a way to do a single function and reuse it for both case? Not need to be part of this PR, but we can have another PR with a re-factoring of this logic.